src/shelly_common.hpp

```diff
@@ -107,4 +107,16 @@ enum class InitialState {
   kMax,
 };
 
+enum class Mode {
```

Comment:
let's have symbolic names for these in js too, a-la Component::Type.
and let's call it something more specific, e.g. `DeviceMode`